lighthouse-core/report/html/renderer/base64.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* global self btoa atob pako CompressionStream Response */
+
+const toBase64 = typeof btoa !== 'undefined' ?
+  btoa :
+  /** @param {string} str */
+  (str) => Buffer.from(str).toString('base64');
+const fromBase64 = typeof atob !== 'undefined' ?
+  atob :
+  /** @param {string} str */
+  (str) => Buffer.from(str, 'base64').toString();
+
+/**
+ * Takes an UTF-8 string and returns a URL-safe base64 encoded string.
```

Comment:
why not?

https://www.example.com/?a=8J+Ygw==

Reply:
I think Brendan is referring to the fact that if the destination were to actually parse these URL fragments according to the spec (i.e. `new URLSearchParams('?foo+bar=1').get('foo+bar')` is `null`, you'd need to do `.get('foo bar')`), things would get weird.

Since we're manually handling all the parsing I don't see it as a major issue, but the comment should probably be updated to indicate this :)

Reply:
yeah, I don't have the `fragment percent-encode set` memorized like @paulirish, but at the very least this isn't what's usually meant by "URL-safe base64", which is https://datatracker.ietf.org/doc/html/rfc4648#section-5

e.g. in node you'd do the explicit `Buffer.from(str).toString('base64url')` dance